tests/clone_panic.rs

```diff
@@ -0,0 +1,65 @@
+extern crate gio;
+extern crate glib;
+extern crate gtk;
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use gio::{ApplicationExt, ApplicationExtManual};
+use gtk::{
+    Application,
+    ApplicationWindow,
+    Button,
+    ButtonExt,
+    ContainerExt,
+    GtkWindowExt,
+    WidgetExt,
+};
+use glib::clone;
+
+struct State {
+    started: bool,
+    count: i32,
+}
+
+impl State {
+    fn new() -> Self {
+        Self {
+            started: false,
+            count: 0,
+        }
+    }
+}
+
+#[test]
+#[should_panic]
```

Comment:
Why will this panic?

Reply:
Because the state will be dropped before the main loop runs.

Reply:
Ah I see. See https://github.com/gtk-rs/glib/pull/527/files#r337428391 then :)